### DISCUSSION TITLE
restore

### DISCUSSION THREAD

**Discussion by Junya98** [pnp/powershell#3016] [Q&A]
I found a script called "Get-PnPRecycleBinItem -RowLimit 500000 | Where {$_.LeafName -like "*.docx"} | Restore-PnpRecycleBinItem -Force". I do not understand. can someone tell me

Should I put the document name in the LeafName part of "Where {$_.LeafName "?

**Comment by AndersRask** [comment]
LeafName is a property on the object. It represents the name of the file or folder that is stored in the recyclebin.

The command you mention will fetch up to 500.000 items with the extension "docx" (word documents) in the first stage recycle bin and restore them without asking for confirmation (-Force)

If you call the first cmdlet alone, you will just show the items in your PowerShell prompt.

I suggest using the Get-Help to fetch examples that might help you:

`Get-Help Restore-PnPRecycleBinItem -ex`
